src/plugins/gui/providers/paybisProvider.ts

```diff
@@ -0,0 +1,470 @@
+            paybisFetch({ method: 'GET', url, path: `v1/currency/pairs`, apiKey })
+              .then(response => {
+                paybisPairs = asPaybisPairs(response)
+              })
+              .catch(e => {
+                console.error(String(e))
+              })
+          ]
+          await Promise.all(promises)
+        }
+
+        if (paybisPairs != null) {
+          for (const paymentMethodPairs of paybisPairs.data) {
+            const { name, pairs } = paymentMethodPairs
+            if (name == null) continue
+            const edgePaymentType = PAYMENT_METHOD_MAP[name]
+            if (edgePaymentType == null) continue
+            for (const pair of pairs) {
+              const { from, to } = pair
+
+              // Add the fiat
+              let paymentMethodObj = allowedCurrencyCodes.buy[edgePaymentType]
+              if (paymentMethodObj == null) {
+                paymentMethodObj = { crypto: {}, fiat: {} }
+                allowedCurrencyCodes.buy[edgePaymentType] = paymentMethodObj
+              }
+              paymentMethodObj.fiat[`iso:${from}`] = true
+
+              // Add the cryptos
+              for (const code of to) {
+                const edgeTokenId = PAYBIS_TO_EDGE_CURRENCY_MAP[code.currencyCode]
+                if (edgeTokenId != null) {
+                  const { pluginId: currencyPluginId } = edgeTokenId
+                  let { currencyCode: ccode } = edgeTokenId
+
+                  if (ccode == null) {
+                    ccode = code.currencyCode
+                  }
+                  // If the edgeTokenId has a tokenId, use it. If not use the currencyCode.
+                  // If no currencyCode, use the key of PAYBIS_TO_EDGE_CURRENCY_MAP
+                  let tokenMap = paymentMethodObj.crypto[currencyPluginId]
+                  if (tokenMap == null) {
+                    tokenMap = {}
+                    paymentMethodObj.crypto[currencyPluginId] = tokenMap
+                  }
+                  tokenMap[ccode] = true
+                }
+              }
+            }
+          }
+        }
+
+        const out = allowedCurrencyCodes[direction][paymentType] ?? { fiat: {}, crypto: {} }
+        return out
+      },
+      getQuote: async (params: FiatProviderGetQuoteParams): Promise<FiatProviderQuote> => {
+        const {
+          amountType,
+          exchangeAmount,
+          regionCode,
+          paymentTypes,
+          pluginId: currencyPluginId,
+          fiatCurrencyCode,
+          displayCurrencyCode,
+          direction,
+          tokenId
+        } = params
+        const paymentType = paymentTypes.find(paymentType => allowedPaymentTypes[direction][paymentType] === true)
+        if (paymentType == null) throw new FiatProviderError({ providerId, errorType: 'paymentUnsupported' })
+
+        const pairs = paybisPairs?.data
+        if (pairs == null) {
+          throw new FiatProviderError({ providerId, errorType: 'assetUnsupported' })
```

Comment:
Seems a odd to be throwing this without comparing to the getQuote params

Reply:
This error implies we were not able to get the supported assets in the first place.

Reply:
Ah right this was the error type that was used as more like a result